actors/builtin/miner/miner_state_test.go

```diff
@@ -440,6 +442,252 @@ func TestVestingFundsStore(t *testing.T) {
+		vspec := &miner.VestSpec{
+			InitialDelay: 10,
+			VestPeriod:   10,
+			StepDuration: 1,
+			Quantization: 2,
+		}
+		vestStart := abi.ChainEpoch(10)
+		vestSum := abi.NewTokenAmount(100)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		// Entry: 22 Funds: 20
+		// Entry: 24 Funds: 20
+		// Entry: 26 Funds: 20
+		// Entry: 28 Funds: 20
+		// Entry: 30 Funds: 20
+
+		totalVested := abi.NewTokenAmount(0)
+		for e := vestStart; e <= vestStart+vspec.VestPeriod+vspec.InitialDelay+1; e++ {
+			amountVested := harness.unlockVestedFunds(e)
+			if e == 23 || e == 25 || e == 27 || e == 29 || e == 31 {
+				assert.Equal(t, abi.NewTokenAmount(20), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			} else {
+				assert.Equal(t, abi.NewTokenAmount(0), amountVested)
+			}
+		}
+		assert.Equal(t, vestSum, totalVested)
+		assert.Zero(t, harness.s.LockedFunds.Int64())
+		assert.True(t, harness.vestingFundsStoreEmpty())
+	})
+
+	t.Run("Vests when quantize, step duration, and vesting period are coprime", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   27,
+			StepDuration: 5,
+			Quantization: 7,
+		}
+		vestStart := abi.ChainEpoch(10)
+		vestSum := abi.NewTokenAmount(100)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		// Entry: 21 Funds: 40
+		// Entry: 28 Funds: 26
+		// Entry: 35 Funds: 26
+		// Entry: 42 Funds: 8
+		totalVested := abi.NewTokenAmount(0)
+		for e := vestStart; e <= 43; e++ {
+			amountVested := harness.unlockVestedFunds(e)
+			switch e {
+			case 22:
+				assert.Equal(t, abi.NewTokenAmount(40), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			case 29:
+				assert.Equal(t, abi.NewTokenAmount(26), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			case 36:
+				assert.Equal(t, abi.NewTokenAmount(26), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			case 43:
+				assert.Equal(t, abi.NewTokenAmount(8), amountVested)
+				totalVested = big.Add(totalVested, amountVested)
+			default:
+				assert.Equal(t, abi.NewTokenAmount(0), amountVested)
+			}
+		}
+		assert.Equal(t, vestSum, totalVested)
+		assert.Zero(t, harness.s.LockedFunds.Int64())
+		assert.True(t, harness.vestingFundsStoreEmpty())
+	})
+
+	t.Run("Vests alternating amounts", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   100,
+			StepDuration: 9,
+			Quantization: 2,
+		}
+		vestStart := abi.ChainEpoch(100)
+		vestSum := abi.NewTokenAmount(1000)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		/*
+			Entry: 110 Funds: 100
+			Entry: 118 Funds: 80
+			Entry: 128 Funds: 100
+			Entry: 136 Funds: 80
+			Entry: 146 Funds: 100
+			Entry: 154 Funds: 80
+			Entry: 164 Funds: 100
+			Entry: 172 Funds: 80
+			Entry: 182 Funds: 100
+			Entry: 190 Funds: 80
+			Entry: 200 Funds: 100
+		*/
```

Comment:
These comments are just saying exactly the same thing as the code, and don't help.